shared/utility/file/fileutil.cpp

```diff
@@ -158,7 +157,7 @@ namespace utility::file {
             // loop through every file within the directory
             for (auto&& file : listDir(directory)) {
                 // specify the correct path within the directory
-                auto path = directory + "/" + file;
+                auto path = directory.append("/").append(file);
```

Comment:
This is introducing a giant bug.
This will modify the directory object and appending that text to it so next time you'll end up with the previous file plus a new file every time.

Really this should be done using the `std::filesystem` api to avoid this problem.
We don't need listFiles or listDir anymore since std::filesystem does it